### PR TITLE
Fix SAML login failure when there is a SP qualifier.

### PR DESCRIPTION
**Pull request by dewniMW**
This PR resolves [1].
[1] https://github.com/wso2/product-is/issues/6287

### CODE REVIEW

File: components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java [hunk 1]
```diff
@@ -1364,7 +1364,7 @@ public static boolean validateXMLSignature(RequestAbstractType request,
 
         }
 
-		
+
```
Comment (GDRDABARERA) [inline components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java]: remove this

File: components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/validators/SPInitSSOAuthnRequestValidator.java [hunk 2]
```diff
@@ -116,18 +107,31 @@ public SAMLSSOReqValidationResponseDTO validate() throws IdentityException {
                 return validationResponse;
             }
 
+            String issuerName = splitAppendedTenantDomain(issuer.getValue());
```
Comment (GDRDABARERA) [inline components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/validators/SPInitSSOAuthnRequestValidator.java]: shall we add a comment here what this do?

File: components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java [hunk 3]
```diff
@@ -2406,6 +2406,27 @@ public static boolean isSAMLNotOnOrAfterPeriodDefined(String sessionNotOnOrAfter
...
+                                SAMLSSOUtil.getTenantDomainFromThreadLocal())) {
+                if (log.isDebugEnabled()) {
+                    String message = "A SAML request with issuer: " + issuer + " is received." +
+                            " A valid Service Provider configuration with the Issuer: " + issuer +
+                            " and Issuer Qualifier: " + issuerQualifier + " is identified by the name: " +
+                            issuerWithQualifier;
+                    log.debug(message);
+                }
+                SAMLSSOUtil.setIssuerWithQualifierInThreadLocal(issuerWithQualifier);
+
```
Comment (nilasini) [inline components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java]: remove extra new line

Comment (dewniMW) [reply components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java]: https://github.com/wso2-extensions/identity-inbound-auth-saml/pull/290

File: components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/validators/SPInitSSOAuthnRequestValidator.java (hunk 2 above)
Comment (dewniMW) [reply components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/validators/SPInitSSOAuthnRequestValidator.java]: https://github.com/wso2-extensions/identity-inbound-auth-saml/pull/290

**Pull request merged**